### PR TITLE
Convert to type-generic math

### PR DESCRIPTION
**Pull request by hannahmbanana**
@Adlai-Holler

### CODE REVIEW

**Comment by hannahmbanana** [comment]
@Adlai-Holler

File: AsyncDisplayKitTests/ASTableViewThrashTests.m [hunk 1]
```diff
@@ -263,7 +263,7 @@ @implementation NSIndexSet (ASThrashHelpers)
 }
 
 /// `insertMode` means that for each index selected, the max goes up by one.
-+ (NSMutableIndexSet *)randomIndexesLessThan:(NSInteger)max probability:(float)probability insertMode:(BOOL)insertMode {
++ (NSMutableIndexSet *)randomIndexesLessThan:(NSInteger)max probability:(CGFloat)probability insertMode:(BOOL)insertMode {
```
Comment (Adlai-Holler) [inline AsyncDisplayKitTests/ASTableViewThrashTests.m]: This one can go back to `float` since it's not a geometry value

**Comment by Adlai-Holler** [comment]
@hannahmbanana Awesome! Could you try building this for a 32-bit device and make sure there are no warnings/errors? iPhone 4 4S 5 5c or iPod touch

**Comment by hannahmbanana** [comment]
4S checked out

**Comment by Adlai-Holler** [comment]
Cool!

**Pull request merged**